sycl/include/CL/sycl/group.hpp

```diff
@@ -435,5 +451,18 @@ template <int Dims> group<Dims> this_group() {
 template <typename Group>
 void group_barrier(Group G, memory_scope FenceScope = Group::fence_scope);
 
+template <>
+inline void group_barrier<group<1>>(group<1>, memory_scope FenceScope) {
+  detail::workGroupBarrier(FenceScope);
+}
+template <>
+inline void group_barrier<group<2>>(group<2>, memory_scope FenceScope) {
+  detail::workGroupBarrier(FenceScope);
+}
+template <>
+inline void group_barrier<group<3>>(group<3>, memory_scope FenceScope) {
+  detail::workGroupBarrier(FenceScope);
+}
+
```

Comment:
Could you please clarify why 3 identical specializations are needed? Can we just provide definition of declaration on line 452?